libs/web/identity/ui/src/lib/web-auth-telegram-button.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useRef } from 'react'
+import { useAppConfig } from '@pubkey-link/web-core-data-access'
+import { createTelegramScriptElement } from './create-telegram-script-element'
+
+declare global {
+  interface Window {
+    TelegramLoginWidget: any;
+    onTelegramAuth?: (user: TelegramUser) => void;
+  }
+}
+
+interface TelegramUser {
+  id: number
+  first_name: string
+  last_name?: string
+  username?: string
+  photo_url?: string
+  auth_date: number
+  hash: string
+}
+
+export function TelegramLoginButton({ onSuccess }: { onSuccess: (data: any) => void }) {
```

Comment:
We can call this method `refresh: () => void` without any properties as it's used to trigger the redirect after authentication.